internal/files/compress.go

```diff
@@ -5,38 +5,32 @@
 package files
 
 import (
-	"compress/flate"
+	"context"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
 	"github.com/elastic/elastic-package/internal/logger"
 
-	"github.com/mholt/archiver/v3"
+	"github.com/mholt/archives"
 )
 
 // Zip function creates the .zip archive from the source path (built package content).
-func Zip(sourcePath, destinationFile string) error {
-	logger.Debugf("Compress using archiver.Zip (destination: %s)", destinationFile)
-
-	z := archiver.Zip{
-		CompressionLevel:       flate.DefaultCompression,
-		MkdirAll:               false,
-		SelectiveCompression:   true,
-		ContinueOnError:        false,
-		OverwriteExisting:      true,
-		ImplicitTopLevelFolder: false,
-	}
+func Zip(ctx context.Context, sourcePath, destinationFile string) error {
```

Comment:
If it is added the context as parameter, it requires to update different functions in the base code.

The main issue is that `elastic-package check` has some difference and it does not have any context. So, it requires to create a new one if that happens:
https://github.com/elastic/elastic-package/pull/2564/files#diff-5794c40aa4c9301aaf9c39eacf7d3c8511f2511cb24bdd95e0a08704ba583a5fR89

Reply:
Another option could be create the context inside this method, so the signature does not change.

Reply:
I think it'd be better to add support for context, it seems to be used in this package for cancellation.